### PR TITLE
Enhancement/experiment controller validators

### PR DESCRIPTION
**Pull request by RidhamShah**
I have added all Experiment related validators and their child Validators (Conditions, Partitions. Factors, ConditionAliases).

### CODE REVIEW

**Comment by RidhamShah** [comment]
@danoswaltCL I am not sure about the `SonarQube` error coming in the PR, can you check it out?

**Comment by danoswaltCL** [comment]
@RidhamShah me neither, it looks like a connection issue. I have disabled sonarqube scan step for now.

File: backend/packages/Upgrade/src/api/DTO/ExperimentDTO.ts [hunk 1]
```diff
@@ -29,63 +44,399 @@ export {
...
+  @IsNotEmpty()
+  @ValidateNested()
+  @Type(() => LevelValidator)
+  public level: LevelValidator;
+}
+
+export class FactorValidator {
+  @IsOptional()
+  @IsString()
+  public id?: string;
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/DTO/ExperimentDTO.ts:94]: Is id optional?

**Comment by VivekFitkariwala** [comment]
Looks good

File: backend/packages/Upgrade/src/api/DTO/ExperimentDTO.ts (hunk 1 above)
Comment (RidhamShah) [reply backend/packages/Upgrade/src/api/DTO/ExperimentDTO.ts:94]: yes

**Pull request merged**